frontend/src/scenes/onboarding/productAnalyticsSteps/DashboardTemplateVariables.tsx

```diff
@@ -0,0 +1,186 @@
+import { IconCheckCircle, IconInfo, IconTrash } from '@posthog/icons'
+import { LemonBanner, LemonButton, LemonCollapse, LemonInput, LemonLabel } from '@posthog/lemon-ui'
+import { useActions, useValues } from 'kea'
+import { useEffect, useState } from 'react'
+import { dashboardTemplateVariablesLogic } from 'scenes/dashboard/dashboardTemplateVariablesLogic'
+import { newDashboardLogic } from 'scenes/dashboard/newDashboardLogic'
+
+import { DashboardTemplateVariableType } from '~/types'
+
+function VariableSelector({
+    variable,
+    hasSelectedSite,
+}: {
+    variable: DashboardTemplateVariableType
+    hasSelectedSite: boolean
+}): JSX.Element {
+    const { activeDashboardTemplate } = useValues(newDashboardLogic)
+    const theDashboardTemplateVariablesLogic = dashboardTemplateVariablesLogic({
+        variables: activeDashboardTemplate?.variables || [],
+    })
+    const { setVariable, resetVariable, goToNextUntouchedActiveVariableIndex, incrementActiveVariableIndex } =
+        useActions(theDashboardTemplateVariablesLogic)
+    const { allVariablesAreTouched, variables, activeVariableIndex } = useValues(theDashboardTemplateVariablesLogic)
+    const [customEventName, setCustomEventName] = useState<string | null>(null)
+    const [showCustomEventField, setShowCustomEventField] = useState(false)
+
+    const FALLBACK_EVENT = {
+        id: '$other_event',
+        math: 'dau',
+        type: 'events',
+    }
+
+    return (
+        <div className="pl-7">
+            <div className="mb-2">
+                <p>
+                    <IconInfo /> {variable.description}
+                </p>
+            </div>
+            {variable.touched && !customEventName && (
+                <div className="flex justify-between items-center bg-bg-3000-light p-2 pl-3 rounded mb-4">
+                    <div>
+                        <IconCheckCircle className="text-success font-bold" />{' '}
+                        <span className="text-success font-bold">Selected</span>
+                        <p className="italic text-muted mb-0">.md-invite-button</p>
+                    </div>
+                    <div>
+                        <LemonButton
+                            icon={<IconTrash />}
+                            type="tertiary"
+                            size="small"
+                            onClick={() => resetVariable(variable.id)}
+                        />
+                    </div>
+                </div>
+            )}
+            {showCustomEventField && (
+                <div className="mb-4">
+                    <LemonLabel>Custom event name</LemonLabel>
+                    <p>
+                        Set the name that you'll use for a custom event (eg. a backend event) instead of selecting an
```

Comment:
What happens if they enter the wrong one - maybe note they can change this later. 

What about supporting multiple events or an action? Future feature?

Reply:
Yeah i think they can do that in the future, this is for someone _just_ getting started, they have no idea what an action is.